### PR TITLE
Move regularize to single eval

### PR DESCRIPTION
**Pull request by hwpang**

### CODE REVIEW

**Comment by hwpang** [comment]
The test is failing and it's because `node.items` contain `Datum` in single eval tree while `node.items` contain `Molecule` for multi eval tree. @mjohnson541 Is there a reason why it's set up this way? I think it makes more sense to me for `node.items` to contain `Datum`.

File: pysidt/utils.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 def data_matches_node(node, data):
-    for m in data:
```
Comment (mjohnson541) [inline pysidt/utils.py:2]: This is another Datum vs Molecule datatype issue

**Review by mjohnson541 — COMMENTED** [review]
Apart from the Datum vs Molecule issue this looks good. 

The issue is that in a Single Eval tree each node has Datum objects at it, but in a Multi Eval tree it has a component of the decomposition of Datum objects, which doesn't have a value so it doesn't make sense to store as a Datum object. We could change the single eval tree to use Molecule objects, which would unify the choice across trees, but it may make some interactions less intuitive in the single eval tree than using Datum objects.

**Comment by hwpang** [comment]
@mjohnson541 ok I fixed the issue. This is ready for another review

**Review by mjohnson541 — APPROVED** [review]

**Pull request merged**